.github/workflows/ci.yml

```diff
@@ -94,36 +100,33 @@ jobs:
 
   build_and_push:
     name: Build docker image
-    needs: [ test, lint ]
+    if: (github.ref == 'refs/heads/main') || (github.event_name == 'release')
```

Comment:
It seems useful to have the Docker build run without merging to main. Can we keep that?